src/v/datalake/coordinator/state_update.cc

```diff
@@ -158,4 +227,77 @@ mark_files_committed_update::apply(topics_state& state) {
     return std::nullopt;
 }
 
+checked<bool, stm_update_error>
+topic_lifecycle_update::can_apply(const topics_state& state) {
+    auto topic_it = state.topic_to_state.find(topic);
+    if (topic_it == state.topic_to_state.end()) {
+        return true;
+    }
+    auto& cur_state = topic_it->second;
+    if (revision < cur_state.revision) {
```

Comment:
We should also think about implementing can_apply as a CAS on the lifecycle state